### ISSUE TITLE
Incorrect --select parameter behavior with error of type E901

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#129]
In GitLab by @vdmit on Aug 25, 2015, 10:03

Seems that --select parameter behavior of flake8 checker is rather weird:

I have sample file:

```
def f()
    pass

```

which by default produces 3 flake8 messages:

`flake8 test.py`

```
test.py:1:9: E901 SyntaxError: invalid syntax
test.py:2:5: E113 unexpected indentation
test.py:3:1: W391 blank line at end of file
````
If I specify `--select=E` option, I see only two of them:

`flake8 --select=E test.py`

```
test.py:2:5: E113 unexpected indentation
test.py:3:1: W391 blank line at end of file
````
Same output is produced with `--select=E,W`.

But, if I specify `--select=E901`, I see NO errors, which is VERY frustrating.
Though, `--select=E113` or `--select=W391` works OK and selects exactly one error message.

Seems that something is wrong with error of type E901.

Note: this is very inconvenient for script testing automation: especially THIS error (and maybe a couple of others) pretend to be the Number One to fix (i.e. be fatal for package build, etc).

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Aug 25, 2015, 10:19

This is a bug with [pep8](https://github.com/PyCQA/pep8), please file it there.

**Comment by asottile** [comment]
In GitLab by @vdmit on Aug 25, 2015, 10:33

This is not correct: pep8 does not produce error of type E901 for this file. It reports only about unexpected indentation.

**Comment by asottile** [comment]
In GitLab by @vdmit on Aug 25, 2015, 10:39

I've played with --select parameter of generic pep8 utility, and I cannot get any problems like this one. All errors/warnings are filtered correctly, even with "patterns" like `--select=E3,W2`.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Aug 25, 2015, 11:22

@vdmit [E901 comes from pep8](https://github.com/PyCQA/pep8/blob/151758ce8f29fd65e10dc0720446c5a3bb164602/pep8.py#L1408).

**Comment by asottile** [comment]
In GitLab by @vdmit on Aug 25, 2015, 11:51

Seems that there are several cases: when it came from pep8, it is processed OK by both flake8 and pep8 utilities. But there are cases when pep8 does not see any errors, and flake8 detects it. And second class of cases is not caught by flake8's `select`.

Consider two samples. First one, which produces E901 both for flake/pep:
```
(

def f():
    test(

```

It produces following errors for `pep8`:
```
test.py:3:1: E122 continuation line missing indentation or outdented
test.py:5:1: W391 blank line at end of file
test.py:5:1: W391 blank line at end of file
test.py:6:1: E901 TokenError: EOF in multi-line statement
```

`pep8 --select=E901` results in just ONE error message, which is an expected one. Here flake acts the same way.

Next, take an example where pep8 is silent - just one line with `=` character:

```
=
```

`pep8` shows nothing here. Though, `flake8` reports one error: `test.py:1:2: E901 SyntaxError: invalid syntax`. And with `--select=E901` it shows NOTHING. I guess it's not acceptable.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Aug 25, 2015, 20:06

So again, I'll repeat, this is in pep8.

There is a difference in behaviour, but again, this is entirely because we rely on pep8's implementation of `--select`. Here's the difference between the two cases:

When running `flake8` without `--select`, `pep8` runs [check_ast](https://github.com/PyCQA/pep8/blob/151758ce8f29fd65e10dc0720446c5a3bb164602/pep8.py#L1510). This (as you can see) causes the `E901 SyntaxError: invalid syntax` that you get as a "duplicate". `check_ast` isn't always called but [generate_tokens](https://github.com/PyCQA/pep8/blob/151758ce8f29fd65e10dc0720446c5a3bb164602/pep8.py#L1522) is always called.

So I'll reiterate once more, file this bug on pep8.
